### PR TITLE
refactor: Update `Window` operation in point free style.

### PR DESCRIPTION
**Pull request by drupol**
This PR:

* [x] Update `Window` in point free style.
* [x] Has unit tests (phpspec)
* [x] Has updated tests
* [x] Has updated documentation
* [x] Is a minor BC break - The behavior has changed when the ```$size``` is zero, which is more in sync with what the Window operation does and avoid mixing types (```T|list<T>``` => ```list<T>```)
  Before: ```Collection::fromIterable(['a', 'b', 'c'])->window(0); // ['a', 'b', 'c']```
  After:  ```Collection::fromIterable(['a', 'b', 'c'])->window(0); // [['a'], ['b'], ['c']]```

### CODE REVIEW

File: spec/loophp/collection/CollectionSpec.php [hunk 1]
```diff
@@ -3999,9 +3999,13 @@ public function it_can_when(): void
...
     {
-        $this::fromIterable(range('a', 'z'))
+        $this::fromIterable(range('a', 'c'))
             ->window(0)
-            ->shouldIterateAs(range('a', 'z'));
+            ->shouldIterateAs([
+                ['a'],
+                ['b'],
+                ['c'],
+            ]);
```
Comment (AlexandruGG) [inline spec/loophp/collection/CollectionSpec.php:4008]: is this change a BC-break? Before `window(0)` would return the items essentially unchanged, but now it puts every element in an array

Comment (drupol) [reply spec/loophp/collection/CollectionSpec.php:4008]: Right, this is indeed a minor BC Break. We can even pass negative numbers and get a different behavior that might be useful. I completely forgot to document that. Thanks.

File: spec/loophp/collection/CollectionSpec.php [hunk 2]
```diff
@@ -4041,111 +4045,13 @@ public function it_can_window(): void
...
-                25 => [
-                    0 => 'x',
-                    1 => 'y',
-                    2 => 'z',
-                ],
+            ]);
+
+        // Unsupported - but tested.
+        $this::fromIterable(range('a', 'e'))
+            ->window(-2)
```
Comment (AlexandruGG) [inline spec/loophp/collection/CollectionSpec.php:4052]: what about the `window(-1)` case? That one is mentioned now in the docs so sounds like it's supported -> we should have a test for it

**Comment by AlexandruGG** [comment]
@drupol I think you merged the wrong one? 😅

**Comment by drupol** [comment]
Oh. Sorry :( Indeed I messed up... too many tabs open here.

**Comment by drupol** [comment]
I'll add a new test in the master branch directly in a few minutes.

**Pull request merged**